lib/hooks/useFormDelayContext.tsx

```diff
@@ -0,0 +1,119 @@
+import { logMessage } from "@lib/logger";
+import { FormElement, FormProperties } from "@lib/types";
+import { createContext, useContext, useState } from "react";
+
+interface FormDelay {
+  startTime: number;
+  requiredQuestions: number;
+}
+
+const timerDefault: FormDelay = {
+  startTime: 0,
+  requiredQuestions: 0,
+};
+
+const FormDelayContext = createContext<{
+  formDelay: FormDelay;
+  setFormDelay: React.Dispatch<React.SetStateAction<FormDelay>>;
+}>({
+  formDelay: timerDefault,
+  setFormDelay: () => {},
+});
+
+export const FormDelayProvider = ({ children }: { children: React.ReactNode }) => {
+  const [formDelay, setFormDelay] = useState(timerDefault);
+  return (
+    <FormDelayContext.Provider value={{ formDelay, setFormDelay }}>
+      {children}
+    </FormDelayContext.Provider>
+  );
+};
+
+// Adds in a little math to make it more unpredictable and calculates the final form delay
+const calculateSubmitDelay = (delayFromFormData: number) => {
+  const secondsBaseDelay = 2; // To help test group forms, make this big e.g. 20000
+  const secondsPerFormElement = 2;
+  return secondsBaseDelay + delayFromFormData * secondsPerFormElement;
+};
+
+export const calculateDelayWithGroups = (
+  startTime: number,
+  endTime: number,
+  requiredQuestions: number
+) => {
+  if (isNaN(startTime) || isNaN(endTime) || isNaN(requiredQuestions)) {
+    return -1;
+  }
```

Comment:
In which scenario we could end up getting inputs that are not valid numbers?

Reply:
I don't think there is a case. But I wanted to make sure, just encase, that this code would not stop a form from submitting. So anywhere there is -1, that will skip the form delay and allow the form to submit right away